pkg/server/jsonrpc.go

```diff
@@ -23,18 +23,24 @@ type JSONRPC interface {
 	communicate(buf []byte) (io.Reader, error)
 }
 
+// TODO: rename to remove unix since now this is generic (last commit)
 type unixSocketJSONRPC struct {
 	transport string
 	socket    *string
 	id        uint64
 }
 
 // NewUnixSocketJSONRPC creates a new instance of JSONRPC which is capable to
-// interact with unix domain socket
+// interact with either unix domain socket, e.g.: /var/tmp/spdk.sock
+// or with tcp connection ip and port tuple, e.g.: 10.1.1.2:1234
 func NewUnixSocketJSONRPC(socketPath string) JSONRPC {
-	// TODO: get transport from socketPath automatically
+	protocol := "tcp"
+	if _, _, err := net.SplitHostPort(socketPath); err != nil {
+		protocol = "unix"
```

Comment:
shoud we check here that `socketPath` exists and raise a panic(return error) if it does not?

Reply:
Maybe… or maybe socket will be created later and then Dial will fail… same with tcp we not checking somebody listening… opened #254 to think about this